docs/user_guide/representing/array.md

```diff
@@ -0,0 +1,463 @@
+
+
+class BannerDoc(BaseDoc):
+    image: ImageUrl
+    title: str
+    description: str
+
+
+class PageDoc(BaseDoc):
+    banner: BannerDoc
+    content: str
+
+
+page1 = PageDoc(
+    banner=BannerDoc(
+        image='https://example.com/image1.png',
+        title='Hello World',
+        description='This is a banner',
+    ),
+    content='Hello wolrd is the most used example in programming, but do you know that ? ...',
+)
+
+page2 = PageDoc(
+    banner=BannerDoc(
+        image='https://example.com/image2.png',
+        title='Bye Bye World',
+        description='This is (distopic) banner',
+    ),
+    content='What if the most used example in programming was Bye Bye World, would programming be that much fun ? ...',
+)
+
+docs = DocList[PageDoc]([page1, page2])
+
+docs.summary()
+```
+
+```cmd
+╭─────── DocList Summary ───────╮
+│                               │
+│   Type     DocList[PageDoc]   │
+│   Length   2                  │
+│                               │
+╰───────────────────────────────╯
+╭────── Document Schema ───────╮
+│                              │
+│   PageDoc                    │
+│   ├── banner: BannerDoc      │
+│   │   ├── image: ImageUrl    │
+│   │   ├── title: str         │
+│   │   └── description: str   │
+│   └── content: str           │
+│                              │
+╰──────────────────────────────╯
+``` 
+
+```python
+print(docs.banner)
+``` 
+
+```cmd
+<DocList[BannerDoc] (length=2)>
+```
+
+Yes, `docs.banner` return a nested `DocList` of `BannerDoc` ! 
+
+You can even access the attribute of the nested `BaseDoc` at the Array level:
+
+```python
+print(docs.banner.url)
+```
+
+```cmd
+['https://example.com/image1.png', 'https://example.com/image2.png']
+```
+
+the same way that with Pydantic and DocArray [BaseDoc][docarray.base_doc.doc.BaseDoc] you would have done:
+
+```python
+print(page1.banner.image)
+```
+
+```cmd
+'https://example.com/image1.png'
+```
+
+### `DocList[DocType]` syntax
+
+As you have seen in the previous section, `AnyDocArray` will expose the same attribute as the `BaseDoc` it contains.
+
+But this concept only work if and only if all of the `BaseDoc` in the `AnyDocArray` have the same schema.
+
+Indeed, if one of your `BaseDoc` have an attribute that the other don't, you will get an error if you try to acces it at
+the array level.
+
+
+!!! note
+    To be able to extend Pydantic API to the Array level, `AnyDocArray` need to contain homogenous Document.
+
+This is where the custom syntax `DocList[DocType]` come into play.
+
+!!!
+    `DocList[DocType]` create a custom [`DocList`][docarray.array.doc_list.doc_list.DocList] that can only contain `DocType` Document.
+
+This syntax is inspired by more statically typed language, and even though it might offend python purist and go against 
+python list principle we believe that it is actually a good user experience to think of Array of `BaseDoc` rather than
+just an array of non-homogenous `BaseDoc`.
```

Comment:
```suggestion
This syntax is inspired by more statically typed languages, and even though it might offend Python purists, we believe that it is actually a good user experience to think of Array of `BaseDoc` rather than
just an array of non-homogenous `BaseDoc`.
```